fairmq/test/parts/_iterator_interface.cxx

```diff
@@ -0,0 +1,81 @@
+/********************************************************************************
+ *    Copyright (C) 2017 GSI Helmholtzzentrum fuer Schwerionenforschung GmbH    *
+ *                                                                              *
+ *              This software is distributed under the terms of the             *
+ *              GNU Lesser General Public Licence (LGPL) version 3,             *
+ *                  copied verbatim in the file "LICENSE"                       *
+ ********************************************************************************/
+
+#include <gtest/gtest.h>
+#include <FairMQParts.h>
+#include <FairMQDevice.h>
+#include <zeromq/FairMQTransportFactoryZMQ.h>
+#include <string>
+#include <sstream>
+#include <algorithm>
+
+namespace
+{
+
+using namespace std;
+
+class RandomAccessIterator : public ::testing::Test {
+  protected:
+    FairMQParts mParts;
+    FairMQTransportFactoryZMQ mFactory;
+
+    RandomAccessIterator()
+    : mParts(FairMQParts{}),
+      mFactory(FairMQTransportFactoryZMQ{})
+    {
+        auto s1 = string{"1"};
+        auto s2 = string{"2"};
+        auto s3 = string{"3"};
```

Comment:
Yes, thinking about it, putting stack data into a message is almost always a bad idea.